mlog/global.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package mlog
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var globalLogger *Logger
+
+func InitGlobalLogger(logger *Logger) {
+	globalLogger = logger
+	Debug = globalLogger.Debug
+	Info = globalLogger.Info
+	Warn = globalLogger.Warn
+	Error = globalLogger.Error
+	Critical = globalLogger.Critical
+}
+
+func RedirectStdLog(logger *Logger) {
+	zap.RedirectStdLogAt(logger.zap.With(zap.String("source", "stdlog")), zapcore.ErrorLevel)
+}
+
+type LogFunc func(string, ...Field)
+
+// DON'T USE THIS Modify the level on the app logger
+func GloballyDisableDebugLogForTest() {
+	globalLogger.consoleLevel.SetLevel(zapcore.ErrorLevel)
+}
+
+// DON'T USE THIS Modify the level on the app logger
+func GloballyEnableDebugLogForTest() {
+	globalLogger.consoleLevel.SetLevel(zapcore.DebugLevel)
+}
+
```

Comment:
Instead of this why not have a function that devs *can* use like:

```go
func DoWithLogLevel(level zapcore.Level, f func()) {
        origLevel := getZapLevel(cfg.LogSettings.ConsoleLevel)
	globalLogger.consoleLevel.SetLevel(level)
	defer globalLogger.consoleLevel.SetLevel(origLevel)
	f()
}
```
```go
mlog.DoWithLogLevel(zapcore.ErrorLevel, func() {
	fmt.Println("Hello")
})
```

Reply:
This is just there to support the existing functions. Hence the "DON'T USE THIS" comment. 
Good idea to add that functionality though for future tests. Feel free to submit that after this gets merged.

Reply:
I can do that.